build/build/src/engine/context.rs

```diff
@@ -414,13 +376,91 @@ impl RunContext {
         }
 
         let enso = BuiltEnso { paths: self.paths.clone() };
+
+        // === Unit tests and Enso tests ===
+        debug!("Running unit tests and Enso tests.");
+        if self.config.test_scala {
+            // Run unit tests
+            sbt.call_arg("set Global / parallelExecution := false; test").await?;
+        }
+        if self.config.test_standard_library {
+            enso.run_tests(IrCaches::No, &sbt, PARALLEL_ENSO_TESTS).await?;
+        }
+        // If we are run in CI conditions and we prepared some test results, we want to upload
+        // them as a separate artifact to ease debugging.
+        if let Some(test_results_dir) = test_results_dir && is_in_env() {
+            // Each platform gets its own log results, so we need to generate unique names.
+            let name = format!("Test_Results_{TARGET_OS}");
+            if let Err(err) = ide_ci::actions::artifacts::upload_compressed_directory(&test_results_dir, name)
+                .await {
+                // We wouldn't want to fail the whole build if we can't upload the test results.
+                // Still, it should be somehow visible in the build summary.
+                ide_ci::actions::workflow::message(MessageLevel::Warning, format!("Failed to upload test results: {err}"));
+            }
+        }
+
+        perhaps_test_java_generated_from_rust_job.await.transpose()?;
+
+        // === Run benchmarks ===
+        debug!("Running benchmarks.");
+        if big_memory_machine {
+            let mut tasks = vec![];
+            // This just compiles benchmarks, not run them. At least we'll know that they can be
+            // run. Actually running them, as part of this routine, would be too heavy.
+            // TODO [mwu] It should be possible to run them through context config option.
+            if self.config.build_benchmarks {
+                tasks.extend([
+                    "runtime/Benchmark/compile",
+                    "language-server/Benchmark/compile",
+                    "searcher/Benchmark/compile",
+                    "std-benchmarks/Benchmark/compile",
+                ]);
+            }
+            // We want benchmarks to run only after the other build tasks are done, as they are
+            // really CPU-heavy.
+            let build_command = (!tasks.is_empty()).then_some(Sbt::concurrent_tasks(tasks));
+            let empty_benchmarks = BTreeSet::new();
+            let benchmark_tasks0 = if TARGET_OS == OS::Windows {
+                &empty_benchmarks
+            } else {
+                &self.config.execute_benchmarks
+            };
```

Comment:
Could we flip this so Mac OSX not running too?

```suggestion
            let benchmark_tasks0 = if TARGET_OS == OS::Linux {
                &self.config.execute_benchmarks
            } else {
                &empty_benchmarks
            };
```

Reply:
Sure, done in c53f5e7a656e6aaeee04637be8a4537179c131ec